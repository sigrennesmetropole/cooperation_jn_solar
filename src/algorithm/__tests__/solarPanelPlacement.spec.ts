import { assert, describe, expect, it } from 'vitest'
import {
  solarPanelPlacement,
  SolarPanelDimension,
  Grid,
} from '../solarPanelPlacement'

describe('solar panel placement', () => {
  it('test', () => {
    const grid: Grid = new Grid([
      [true, true, true, false],
      [true, true, true, true],
      [true, true, false, true],
    ])

    const solarPanelDimension: SolarPanelDimension = { length: 2, width: 2 }
    solarPanelPlacement(grid, solarPanelDimension)
    assert.equal(Math.sqrt(4), 2)
  })

  it('bar', () => {
    expect(1 + 1).eq(2)
  })

  it('snapshot', () => {
    expect({ foo: 'bar' }).toMatchSnapshot()
  })
})
