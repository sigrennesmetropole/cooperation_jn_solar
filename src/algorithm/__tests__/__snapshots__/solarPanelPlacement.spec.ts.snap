// Vitest Snapshot v1

exports[`solar panel placement > snapshot 1`] = `
{
  "foo": "bar",
}
`;
